src/main/kotlin/com/sourcegraph/cody/config/CodyPersistentAccountsHost.kt

```diff
@@ -22,12 +21,11 @@ class CodyPersistentAccountsHost(private val project: Project?) : CodyAccountsHo
       // Notify Cody Agent about config changes.
       CodyAgent.getServer(project)
           ?.configurationDidChange(ConfigUtil.getAgentConfiguration(project))
-      val codyToolWindowContent = CodyToolWindowContent.getInstance(project)
-      codyToolWindowContent.refreshPanelsVisibility()
-      codyToolWindowContent.embeddingStatusView.updateEmbeddingStatus()
-      UpgradeToCodyProNotification.autocompleteRateLimitError.set(null)
-      UpgradeToCodyProNotification.chatRateLimitError.set(null)
-      CodyAutocompleteStatusService.resetApplication(project)
+
```

Comment:
Is `addAccount` called every time plugin is started?
Previously that logic was also called if user was already logged in, and pans was created for the first time.
How will it work now if it will be only called on account settings change?

Reply:
I did a quick search and it seems `addAccount` is not called unless a user performs the login action.

> Previously that logic was also called if user was already logged in, and pans was created for the first time.

I do not understand. The whole reset logic is called here as before but from `AccountSettingChangeListener`. Here, from `addAccount` and there, from `AccountConfigurable.kt::apply`.

Reply:
Yes, I'm talking about `AccountConfigurable.kt::apply.`.
That function was called in various situations, e.g.:

```
    if (connectionAndAuthState == ConnectionAndAuthState.COULD_NOT_CONNECT) {
      // ...
      emptyText.appendLine(
          "Click here to configure your Sourcegraph Cody + Code Search settings.",
          new SimpleTextAttributes(STYLE_PLAIN, JBUI.CurrentTheme.Link.Foreground.ENABLED),
          __ ->
              ShowSettingsUtil.getInstance()
                  .showSettingsDialog(project, AccountConfigurable.class));

```

If I understand this code correctly clicking on that link will create panel which in turn will instantiate `AccountConfigurable`. Previously it was causing reset logic to be executed, and now I think it's not anymore?

Reply:
Resolved offline, at the beginning of the `apply` method there is notification to the listener, so everything should work as expected.